DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_packets.py

```diff
@@ -0,0 +1,93 @@
+            {"port": 1, "settings": [{"vlan": 1, "untagged": True, "pvid": True}]},
+            {"port": 2, "settings": [{"vlan": 1, "untagged": True, "pvid": True}]},
+            {"port": 3, "settings": [{"vlan": 1, "untagged": False, "pvid": True}]},)
+
+
+async def test_vlan_tagged_packet_size(testbed):
+    """
+    Test Name:VLAN tagged packet size
+    Test Suite: suite_functional_vlan
+    Test Overview: Test packet with VLAN tag on untagged port is bigger by 4 bytes
+    Test Procedure:
+    1. Initiate test params.
+    2. Set links to vlans.
+    3. Map receiving and non receiving dut_ports.
+        Port 1 ->  tx_port
+        Ports 2, 3, 4 -> rx_ports
+    4. Setup packet stream(s) for the broadcast packet:
+    5. Send traffic to rx_ports
+    6. Verify that the tagged packet size is bigger in 4 bytes than untagged packet
+    """
+
+    # 1. Initiate test params.
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgn connections")
+        return
+    device = dent_devices[0].host_name
+    tg_ports = tgen_dev.links_dict[device][0]
+    dut_ports = tgen_dev.links_dict[device][1]
+    tagged_traffic_port = 3
+
+    await configure_bridge_setup(device, dut_ports, default_pvid=1)
+
+    # 2. Set links to vlans.
+    await configure_vlan_setup(device, port_map, dut_ports)
+
+    # 3.Map receiving and non receiving dut_ports.
+    dev_groups = tgen_utils_dev_groups_from_config([
+        {"ixp": tg_ports[0], "ip": "100.1.1.2", "gw": "100.1.1.6", "plen": 24, },
+        {"ixp": tg_ports[1], "ip": "100.1.1.3", "gw": "100.1.1.6", "plen": 24, },
+        {"ixp": tg_ports[2], "ip": "100.1.1.4", "gw": "100.1.1.6", "plen": 24, },
+        {"ixp": tg_ports[3], "ip": "100.1.1.5", "gw": "100.1.1.6", "plen": 24, }])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, dut_ports, dev_groups)
+
+    tx_ports = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = [dev_groups[tg_ports[1]][0]["name"],
+                dev_groups[tg_ports[2]][0]["name"],
+                dev_groups[tg_ports[3]][0]["name"]]
+
+    # 4. Setup packet stream(s).
+    streams = {f"Untagged traffic": {
+        "type": "raw",
+        "protocol": "802.1Q",
+        "ip_source": tx_ports,
+        "ip_destination": rx_ports,
+        "srcMac": "02:00:00:00:00:01",
+        "dstMac": "ff:ff:ff:ff:ff:ff",
+        "frameSize": 512
+    }}
+
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+    # 5. Send traffic to rx_ports.
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(5)
+    await tgen_utils_stop_traffic(tgen_dev)
+
+    # 6. Verify that the tagged packet size is bigger in 4 bytes than untagged packet
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+    expected_packet_size = streams["Untagged traffic"].get("frameSize")
```

Comment:
"actual" and "expected" are a little bit confusing. Especially when you compare with expected + 4. I suggest to use tx_frame_size and rx_frame_size.